### ISSUE TITLE
Roadmap of FastReID

### ISSUE THREAD

**Issue by L1aoXingyu** [JDAI-CV/fast-reid#393]
We keep this issue open to collect feature requests from users and hear your voice. Our monthly release plan is available here.

You can either:
1. Suggest a new feature by leaving a comment.
2. Vote for a feature request with 👍 or be against with 👎. (Remember that developers are busy and cannot respond to all feature requests, so vote for your most favorable one!)
3. Tell us that you would like to help implement one of the features in the list or review the PRs. (This is the greatest thing to hear about!)
## V1.4

- [ ] fused LAMB optimizer
- [ ] ZeRO optimizer
- [ ] Gradient accumulation
- [ ] Swin Transformer backbone

## V1.3 (May)

- [x] Clip gradient (07b8251ccb9f296e4be5d49fc98efebaa12d679f)
- [x] Vision Transformer backbone (2cabc3428adf5cb3fc0be0f273cc025c4b0a8781)
- [x] [prefetch_generator](https://github.com/justheuristic/prefetch_generator) (#456)
- [x] Reduce evaluation memory cost

## V1.2 (March)

- [x] Multiple machine training (#425)
- [x] Torch2trt pipeline (#428) 
- [x] [RepVGG](https://github.com/DingXiaoH/RepVGG) backbone (#429)
- [x] [Partial FC](https://github.com/deepinsight/insightface/tree/master/recognition/partial_fc/pytorch)(https://github.com/JDAI-CV/fast-reid/commit/44cee30dfc929df2051d1ca56e36dc152866e96b)
- [ ] Visualize activation maps

## V1.1 (February)

- [ ] Documents
- [x] ~~[RepVGG](https://github.com/DingXiaoH/RepVGG) backbone (#429)~~(delayed to V.12)
- [x] ~~Torch2trt pipeline. (#428)~~(delayed to V1.2)
- [x] NAIC20 winner solution
- [x] Multi-teacher KD
- [x] ~~[Partial FC](https://github.com/deepinsight/insightface/tree/master/recognition/partial_fc/pytorch)(https://github.com/JDAI-CV/fast-reid/commit/44cee30dfc929df2051d1ca56e36dc152866e96b)~~ (delayed to V1.2)
- [x] reid model with tensorrt network definition APIs

**Comment by 404289680** [comment]
Partial FC 👍

**Comment by mingde510** [comment]
RepVGG backbone

**Comment by douzi0248** [comment]
NAIC20 winner solution

**Comment by F0xZz** [comment]
reid model with tensorrt network definition APIs 

![](https://github.githubassets.com/images/icons/emoji/unicode/1f44d.png)

**Comment by F0xZz** [comment]
reid model with tensorrt network definition APIs ![image](https://user-images.githubusercontent.com/41953310/105965289-2ffb8e80-60be-11eb-8453-fe9e646de453.png)

**Comment by TCHeish** [comment]
reid model with tensorrt network definition APIs  :thumbup:  I will PR soon!

**Comment by raymondchu-z** [comment]
visualize activation maps

**Comment by TelBotDev** [comment]
In view of the standardization of this project is done very well, at the same time there are more and more users, I want to request the project to pin one issue just next "Roadmap of FastReID" and "Official wechat group". That is, if you think you can get same or even better results under different configuration, then you can share the configuration file.

The reason for this is that I cannot get the same result because I do not have 4 NVIDIA P40 GPU :) [here](https://github.com/JDAI-CV/fast-reid/blob/master/MODEL_ZOO.md#vehicleid-baseline), with many times adjusting the learning rate or some else.

**Comment by L1aoXingyu** [comment]
@TelBotDev Great suggestion!

**Comment by DaBaiHao** [comment]
Great job

**Comment by stoneyang** [comment]
Vision Transformer backbone :+1:

**Comment by edwardnguyen1705** [comment]
Unsupervised Pre-training:
`
https://github.com/DengpanFu/LUPerson
`

**Comment by edwardnguyen1705** [comment]
Join training with attribute loss if available

**Comment by Thangbluee** [comment]
OSNET backbone please.

**Comment by edwardnguyen1705** [comment]
Batch DropBlock
https://arxiv.org/abs/1811.07130

**Comment by edwardnguyen1705** [comment]
CGD
https://arxiv.org/abs/1903.10663

**Comment by hardik0** [comment]
OpenVINO support

**Comment by wsqshiqing** [comment]
Make Loss can be config like backbone and head
Also, Make Dataset can be config like Detectron2

**Comment by shen338** [comment]
Any ETA on swin transformer backbone?
